contracts/EverlongStrategy.sol

```diff
@@ -0,0 +1,482 @@
+
+    /// @dev Structure to store and account for everlong-controlled positions.
+    Portfolio.State internal _portfolio;
+
+    // ╭───────────────────────────────────────────────────────────────────────╮
+    // │                              CONSTRUCTOR                              │
+    // ╰───────────────────────────────────────────────────────────────────────╯
+
+    /// @notice Creates a new EverlongStrategy.
+    /// @param _asset Asset to use for the strategy.
+    /// @param __name Name for the strategy.
+    /// @param _hyperdrive Address of the Hyperdrive instance.
+    /// @param _asBase Whether `_asset` is Hyperdrive's base asset.
+    constructor(
+        address _asset,
+        string memory __name,
+        address _hyperdrive,
+        bool _asBase
+    ) BaseStrategy(_asset, __name) {
+        hyperdrive = _hyperdrive;
+        asBase = _asBase;
+        _poolConfig = IHyperdrive(_hyperdrive).getPoolConfig();
+    }
+
+    // ╭───────────────────────────────────────────────────────────────────────╮
+    // │                          STRATEGY OVERRIDES                           │
+    // ╰───────────────────────────────────────────────────────────────────────╯
+
+    /// @inheritdoc BaseStrategy
+    function _deployFunds(uint256) internal override {
+        // Do nothing.
+        // Opening longs on Hyperdrive is sandwichable so funds should only be
+        // deployed when the `keeper` calls `tend()`.
+        return;
+    }
+
+    /// @inheritdoc BaseStrategy
+    function _freeFunds(uint256 _amount) internal override {
+        // Close all matured positions (if any).
+        // TODO: Determine whether `_tendConfig.positionClosureLimit` should
+        //       affect this.
+        uint256 output = _closeMaturedPositions(0);
+
+        // Close immature positions if additional funds need to be freed.
+        if (_amount > output) {
+            _closePositions(_amount - output);
+        }
+    }
+
+    /// @inheritdoc BaseStrategy
+    function _harvestAndReport()
+        internal
+        override
+        returns (uint256 _totalAssets)
+    {
+        // If the strategy isn't shut down, call `_tend()` to close mature
+        // positions and spend idle if needed.
+        if (!TokenizedStrategy.isShutdown() && _tendTrigger()) {
+            _tend(ERC20(asset).balanceOf(address(this)));
+        }
+
+        // Recalculate the value of assets the strategy controls.
+        _totalAssets = calculateTotalAssets();
+    }
+
+    /// @inheritdoc BaseStrategy
+    function availableDepositLimit(
+        address
+    ) public view override returns (uint256) {
+        // Limit deposits to the maximum long that can be opened in hyperdrive.
+        return IHyperdrive(hyperdrive).calculateMaxLong();
+    }
+
+    /// @inheritdoc BaseStrategy
+    function _tendTrigger() internal view override returns (bool) {
+        return hasMaturedPositions() || canOpenPosition();
+    }
```

Comment:
Some docs explaining what this is inline would be good. I'll defer to you on style in the Everlong repo, but I'm not a fan of heavy `inheritdoc` personally. I prefer the documentation to live in-line with the code, and I'm okay sacrificing a bit of comment duplication for that. 

I get the argument for `inheritdoc` for things like docs generation and even working in an editor with good LSP support, but we don't have those tools available during reviews.

Reply:
agreed and updated :)